streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1343,6 +1355,53 @@ private long pollPhase() {
         return pollLatency;
     }
 
+    private void handleAssignmentFromStreamsRebalanceProtocol() {
+        // Process assignment from Streams Rebalance Protocol
+        Assignment newAssignment = null;
+        synchronized (streamsAssignmentInterface.targetAssignment) {
+            if (!streamsAssignmentInterface.targetAssignment.equals(streamsAssignmentInterface.reconciledAssignment)) {
+                newAssignment = streamsAssignmentInterface.targetAssignment.copy();
+            }
+        }
+        if (newAssignment != null) {
+
+            final Map<TaskId, Set<TopicPartition>> activeTasksWithPartitions =
+                pairWithTopicPartitions(newAssignment.activeTasks.stream());
+            final Map<TaskId, Set<TopicPartition>> standbyTasksWithPartitions =
+                pairWithTopicPartitions(Stream.concat(newAssignment.standbyTasks.stream(), newAssignment.warmupTasks.stream()));
+
+            log.info("Processing new assignment {} from Streams Rebalance Protocol", newAssignment);
+
+            taskManager.handleAssignment(
```

Comment:
Don't you think we also need to revoke partitions?
Without revocation we do not commit the tasks that are migrated away.

Reply:
Yeah... good point. I'll try to fix it. I don't have a very good way to figure out what we need to do here, so this is very helpful

Reply:
No wait. Here was my thinking. `onPartitionsRevoked` is invoked by the `MembershipManager`, as well as `onPartitionsLost` and `onPartitionsAssigned`. So the only thing we need to "replace" right now is `StreamsPartitionsAssignor.onAssignment`.

However, the order isn't properly maintained.

In the original coordinator, we have `onPartitionsRevoked` -> `onAssignment` -> `onPartitionsAssigned`, but we don't guarantee the order in "streams protocol client". So yeah, that's a concern. I wonder if we can get the POC working without this.

This was what was making me think that we should also rewrite `MembershipManager`, at which point we don't use any of the group-management facilities besides commiting from the normal consumer